lib/global_uid/base.rb

```diff
@@ -29,167 +20,95 @@ def self.create_uid_tables(id_table_name, options={})
       type     = options[:uid_type] || "bigint(21) UNSIGNED"
       start_id = options[:start_id] || 1
 
-      engine_stmt = "ENGINE=#{global_uid_options[:storage_engine] || "MyISAM"}"
+      engine_stmt = "ENGINE=#{GlobalUid.configuration.storage_engine}"
 
-      with_connections do |connection|
-        connection.execute("CREATE TABLE IF NOT EXISTS `#{id_table_name}` (
+      with_servers do |server|
+        server.connection.execute("CREATE TABLE IF NOT EXISTS `#{id_table_name}` (
         `id` #{type} NOT NULL AUTO_INCREMENT,
         `stub` char(1) NOT NULL DEFAULT '',
         PRIMARY KEY (`id`),
         UNIQUE KEY `stub` (`stub`)
         ) #{engine_stmt}")
 
         # prime the pump on each server
-        connection.execute("INSERT IGNORE INTO `#{id_table_name}` VALUES(#{start_id}, 'a')")
+        server.connection.execute("INSERT IGNORE INTO `#{id_table_name}` VALUES(#{start_id}, 'a')")
       end
     end
 
     def self.drop_uid_tables(id_table_name)
-      with_connections do |connection|
-        connection.execute("DROP TABLE IF EXISTS `#{id_table_name}`")
+      with_servers do |server|
+        server.connection.execute("DROP TABLE IF EXISTS `#{id_table_name}`")
       end
     end
 
-    def self.new_connection(name, connection_timeout)
-      raise "No id server '#{name}' configured in database.yml" unless ActiveRecord::Base.configurations.to_h.has_key?(name)
-      config = ActiveRecord::Base.configurations.to_h[name]
-      c = config.symbolize_keys
-
-      raise "No global_uid support for adapter #{c[:adapter]}" if c[:adapter] != 'mysql2'
-
-      begin
-        Timeout.timeout(connection_timeout, ConnectionTimeoutException) do
-          ActiveRecord::Base.mysql2_connection(config)
-        end
-      rescue ConnectionTimeoutException => e
-        notify e, "Timed out establishing a connection to #{name}"
-        nil
-      rescue Exception => e
-        notify e, "establishing a connection to #{name}: #{e.message}"
-        nil
+    def self.init_server_info
+      id_servers = GlobalUid.configuration.id_servers
+      increment_by = GlobalUid.configuration.increment_by
+      connection_timeout = GlobalUid.configuration.connection_timeout
+
+      id_servers.map do |name|
+        GlobalUid::Server.new(name,
+          increment_by: increment_by,
+          connection_timeout: connection_timeout
+        )
```

Comment:
This refactor is ideal to accept the `increment_offset` value for a given server!

Reply:
Agreed. As [mentioned in #63](https://github.com/zendesk/global_uid/pull/63#discussion_r411862006) I'll look updating the configuration to receive a hash of ID servers with their respective increment_by and increment_offset settings, validating them. I think the change deserves its own PR